### ISSUE TITLE
feature to add access token saved in Mongo

### ISSUE THREAD

**Issue by dotdreaming** [Receiptful/node-bigcommerce#5]
Hello,

This isn't an issue but a feature request.

It would be great if node-bigcommerce could have an option to save the access object to a mongodb db. This is the only thing I am missing and I am struggling to get this working with my app.

Great module :)

Many thanks

**Comment by RickCraig** [comment]
Hello,

I apologise for replying 2 weeks later, I did have this on my todo list, but I haven't got around to replying until now.

Maybe I can give you a helping hand in saving the access object to your MongoDB (It's exactly the same database we use);

Maybe if I show you what I did, it will help with what you are doing:

### Simple Mongoose Model
```javascript
var mongoose = require('mongoose'),
  Schema = mongoose.Schema;

var userSchema = new Schema({
  email: String,
  store: {
    domain: String,
    token: String,
    scope: String,
    context: String,
    user: {
      id: Number,
      email: String,
      username: String
    },
    installed: { type: Boolean, default: true }
  }
});

mongoose.model('User', userSchema);
```

### Authorise a user
```javascript
var BigCommerce = require('node-bigcommerce'),
  helpers = require('/path/to/helpers');

// Instantiate the Bigcommerce object
var bc = new BigCommerce({
  logLevel: 'error',
  clientId: 'bigcommerce_clientid',
  secret: 'bigcommerce_secret',
  callback: 'http://myurl/path/to/callback',
  responseType: 'json'
});

// Inside your route (or wherever you are calling the auth function)
bc.authorise(function(err, data) {
  // Handle any errors that are returned
  if (err) return throw err;
  
  // If no user is returned handle it (should never happen)
  if(!data || !data.user) {
    return throw new Error('Data returned from the authorisation ' +
          'process was either missing the data or user');
  }

  helpers.registerUser(data, function(err) {
    if (err) return throw err;
    // Carry on with your process of registration or finish it here
    // at this point (using express) we would render the success
    // auth page
  });
}):
```

### Register User
We have this in a helpers js file.
```javascript
var mongoose = require('mongoose'),
  BigCommerce = require('node-bigcommerce');

var User = mongoose.model('User');
var bc = new BigCommerce({
  logLevel: 'error',
  clientId: 'bigcommerce_clientid',
  secret: 'bigcommerce_secret',
  callback: 'http://myurl/path/to/callback',
  responseType: 'json'
});

var populateStore = function(accessToken, context, callback) {
  bc.config.accessToken = accessToken;
  bc.config.storeHash = context.split('/')[1];
  bc.config.responseType = 'json';

  bc.get('/store', function(err, data) {
    if (err) {
      // Log the error if you want...
      return callback(null);
    }
   callback(data);
  });
};

var registerUser = function(data, callback) {

  // At this point we would get the store details
  populateStore(data.access_token, data.context, function(store) {
    var user = new User({
      email: data.user.email,
      bigcommerce: {
        domain: store.domain,
        token: data.access_token,
        scope: data.scope,
        context: data.context,
        user: {
          id: data.user.id,
          email: data.user.email,
          username: data.user.username
        }
    };

    // Save the user and move on to the next task
    user.save(callback);
  });
};

module.exports = {
 registerUser: registerUser
};
```

### TL:DR
We take the data object returned by ```bc.authorise``` and send it to get the store's information. You may not need this stage. Once we collect the store data, we put the access object into a new user object and save it. Your system will most likely be different than this, but the underlying process to save the object should be the same.

I would love to hear back if this helped you out (or not of course). 

Thank you for the positive feedback on the module :smiley: 
Rick

**Comment by dotdreaming** [comment]
Hi Rick,
This is brilliant! You have been very helpful. I have got my app storing the access_token in mongo. I am not storing all of the additional store information like you are but I have it working  :)
I see you have a field to state if the app is installed or not. That is interesting and a good idea. So you don't delete the user from DB on uninstall, you just change the "installed" field to false.  Is that right?

Also, I see you call a new BigCommerce instance (var bc = new BigCommerce({}) ) in both the Authorise and the register user code. I have duplicate calls of this too. Is there a way of just calling this once and requiring it for the other times we need to use it?

Kind regards,
Brian

**Comment by RickCraig** [comment]
That should be reasonably easy to do by exporting the instance from the helpers and using it in the scripts that require/import it, like so:

helpers.js
```javascript
var BigCommerce = require('node-bigcommerce');

var bc = new BigCommerce({
  logLevel: 'error',
  clientId: 'bigcommerce_clientid',
  secret: 'bigcommerce_secret',
  callback: 'http://myurl/path/to/callback',
  responseType: 'json'
});

/**
 * All your helpers like the comments above
 */

module.exports = {
  bc: bc
};
```

stuff.js
```javascript
var helpers = require('/path/to/helpers');

helpers.bc.authorise(function(err, data) {
  // Your script for handling the authorisation success
}):
```

That will take away the duplication.

Thanks,
Rick

**Comment by dotdreaming** [comment]
Perfect. I will give that a go :)
thank you again

**Comment by NaveenaTsMaximizer** [comment]
Hi RickCraig,
Is there any example to store access token in MongoBd? If you have can you provide me
